js/mapModel.js

```diff
@@ -47,130 +47,49 @@ class mapModel{
     initTileMeadow(x, y){
         this.addTile(x, y, 0, Meadow, '');
     }
-    setVisible(heroX, heroY){
-        this.getTile(heroX, heroY).Visited = 1; // 1 is visible
-        //REVEALS TILE TO THE RIGHT
-        if(heroX + 1 < this.mapSize){
-            this.getTile(heroX + 1, heroY).Visited = 1;
-            //WRAPS RIGHT TILE IF HERO IS ON FAR RIGHT EDGE OF MAP
-        }else if(heroX + 1 >= this.mapSize){
-            this.getTile(0, heroY).Visited = 1;
-        }
-        //REVEALS TILE TO LEFT OF HERO
-        if(heroX - 1 >= 0){
-            this.getTile(heroX - 1, heroY).Visited = 1;
-            //WRAPS LEFT TILE IF HERO IS ON FAR LEFT EDGE OF MAP
-        }else if(heroX -1 < 0){
-            this.getTile(this.mapSize - 1, heroY).Visited = 1;
-        }
-        //REVEALS TILE ABOVE HERO
-        if(heroY + 1 < this.mapSize){
-            this.getTile(heroX, heroY + 1).Visited = 1;
-            //WRAPS TILE ABOVE HERO TO THE BOTTOM IF HERO IS ON TOP EDGE OF MAP
-        }else if(heroY + 1 >= this.mapSize){
-            this.getTile(heroX, 0).Visited = 1;
-        }
-        //REVEALS TILE BELOW HERO
-        if(heroY - 1 >= 0){
-            this.getTile(heroX, heroY - 1).Visited = 1;
-            //WRAPS TILE BELOW HERO TO THE TOP IF THE HERO IS ON THE BOTTOM EDGE OF THE MAP
-        }else if(heroY - 1 < 0){
-            this.getTile(heroX, this.mapSize -1).Visited = 1;
-        }
-        //REVEALS TILE TO THE TOP RIGHT OF HERO
-        if(heroX + 1 < this.mapSize && heroY + 1 < this.mapSize){
-            this.getTile(heroX + 1, heroY + 1).Visited = 1;
-            //REVEALS 0,0 IF HERO IS IN TOP RIGHT CORNER OF MAP
-        }else if(heroX + 1 >= this.mapSize && heroY + 1 >= this.mapSize){
-            this.getTile(0 ,0).Visited = 1;
-            //WRAPS TILE TO BOTTOM
-        }else if(heroX + 1 < this.mapSize && heroY + 1 >= this.mapSize){
-            this.getTile(heroX + 1, 0).Visited = 1;
-            //WRAPS TILE TO OTHER SIDE
-        }else if(heroX + 1 >= this.mapSize && heroY + 1 < this.mapSize){
-            this.getTile(0, heroY +1).Visited = 1;
-        }
 
-        //REVEALS TILE TO THE TOP LEFT OF HERO
-        if(heroX - 1 >= 0 && heroY + 1 < this.mapSize){
-            this.getTile(heroX - 1, heroY + 1).Visited = 1;
-            //REVEALS mapSize - 1, 0 IF HERO IS IN TOP LEFT CORNER
-        }else if(heroX - 1 < 0 && heroY + 1 >= this.mapSize){
-            this.getTile(this.mapSize -1, 0).Visited = 1;
-            //WRAPS TILE TO BOTTOM
-        }else if(heroX - 1 > 0 && heroY + 1 >= this.mapSize){
```

Comment:
As an example of a bug in the repeated those, this should be ">= 0". It's very hard to get it right every time in repetitive code.